### PR TITLE
EOS-19737 : Added Reset/Cleanup for utils_setup

### PR DESCRIPTION
**Pull request by vastradparayya**
This provides support for Reset/Cleanup for utils_setup
Associated jira - https://jts.seagate.com/browse/EOS-19737

Signed-off-by: Parayya Vastrad <parayya.vastrad@seagate.com>

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: py-utils/src/setup/utils.py [hunk 1]
```diff
@@ -160,5 +159,31 @@ def test():
 
     @staticmethod
     def reset():
-        """ Performs Configuraiton reset """
+        """ Remove/Delete all the data that was created after post install """
+
+        import os
+        conf_file = "/etc/cortx/message_bus.conf"
+        if not os.path.exists(conf_file):
+            raise SetupError(errno.ENOENT, "Reset/Cleanup has already done or config file not found! ")
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:167]: We should not throw error in this case. If someone accidently deletes this file, you will never cleanup message bus.

File: py-utils/src/setup/utils.py [hunk 2]
```diff
@@ -160,5 +159,31 @@ def test():
...
+            mb.deregister_message_type(message_types_list)
+        return 0
+
+    @staticmethod
+    def cleanup():
+        """ Cleanup message bus config and logs. """
+        import os
+        config_file = "/etc/cortx/message_bus.conf"
+        if not os.path.exists(config_file):
+            raise SetupError(errno.ENOENT, "Cleanup has already done or config file not found! ")
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:183]: Same case here as well. Reset / Cleanup and other ops are idempotent ops i.e. they can be run multiple times and must not throw error.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils.py (hunk 1 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done. Made the changes

File: py-utils/src/setup/utils.py (hunk 2 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done. Made the changes

File: py-utils/src/setup/utils_setup.py [hunk 3]
```diff
@@ -158,7 +158,8 @@ def __init__(self, args: dict):
         super().__init__(args)
 
     def process(self):
-        return 0
+        rc = Utils.cleanup()
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils_setup.py:161]: Why not call validate here ? like we do for reset ?

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils.py [hunk 4]
```diff
@@ -160,5 +159,32 @@ def test():
...
+    def cleanup():
+        """ Cleanup message bus config and logs. """
+        import os
+        config_file = "/etc/cortx/message_bus.conf"
+        if os.path.exists(config_file):
+            # delete data/config stored
+            try:
+                os.remove(config_file)
+            except OSError as err:
+                raise SetupError(errno.EINTR, "Error deleting config file %s, %s", config_file, err)
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:189]:
```
except OSError as e:
    raise SetupError(e.errno, "Error deleting config file %s, %s", config_file, e)
```

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils_setup.py (hunk 3 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils_setup.py:161]: As of now we are not doing validation. Utils.validate('reset'), it has empty implementation

File: py-utils/src/setup/utils.py [hunk 5]
```diff
@@ -160,5 +159,32 @@ def test():
...
+            # delete message_types
+            from cortx.utils.message_bus.message_bus_client import MessageBusAdmin
+            from cortx.utils.message_bus.error import MessageBusError
+            try:
+                mb = MessageBusAdmin("reset_admin_id")
+                message_types_list = mb.list_message_types()
+                if message_types_list:
+                    mb.deregister_message_type(message_types_list)
+            except Exception as err:
+                raise MessageBusError(errno.ETIMEDOUT, "Error while resetting! %s", err)
```
[the remark this answers is not in the document]
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done. Made the changes

File: py-utils/src/setup/utils.py (hunk 4 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done. Made the changes

File: py-utils/src/utils/errors.py [hunk 6]
```diff
@@ -17,6 +17,7 @@
 
 OPERATION_SUCESSFUL = 0x0000
 INTERNAL_ERROR = 0x1005
+OP_FAILED = 200
```
Comment (ujjwalpl) [inline py-utils/src/utils/errors.py:20]: 200 was just an example. We need to pick up a number in the series like 0x1006
```
INTERNAL_ERROR = 0x1005
OP_FAILED = 0x1006
```

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils.py [hunk 7]
```diff
@@ -160,5 +161,31 @@ def test():
...
+        conf_file = "/etc/cortx/message_bus.conf"
+        if os.path.exists(conf_file):
+            # delete message_types
+            from cortx.utils.message_bus.message_bus_client import MessageBusAdmin
+            try:
+                mb = MessageBusAdmin("reset_admin_id")
+                message_types_list = mb.list_message_types()
+                if message_types_list:
+                    mb.deregister_message_type(message_types_list)
+            except Exception as e:
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:176]: This does not follow the code I suggested. There are still errror. Message Bus throws MessageBusError. You must catch it.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils.py (hunk 7 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py:179]: Done.

File: py-utils/src/utils/errors.py (hunk 6 above)
Comment (vastradparayya) [reply py-utils/src/utils/errors.py]: Done

File: py-utils/src/setup/utils.py [hunk 8]
```diff
@@ -160,5 +162,33 @@ def test():
...
+        if os.path.exists(conf_file):
+            # delete message_types
+            from cortx.utils.message_bus.message_bus_client import MessageBusAdmin
+            try:
+                mb = MessageBusAdmin("reset_admin_id")
+                message_types_list = mb.list_message_types()
+                if message_types_list:
+                    mb.deregister_message_type(message_types_list)
+            except MessageBusError as e:
+                raise SetupError(errno.ETIMEDOUT, "Can not reset Message Bus. %s", e)
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:178]: raise SetupError(e.rc, "Can not reset Message Bus. %s", e)

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils.py (hunk 8 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Sorry Ujjwal, got it.
Done

**Review by ujjwalpl — APPROVED** [review]

**Review by rahul-tripathi-git — COMMENTED** [review]
+1

**Review by rahul-tripathi-git — APPROVED** [review]
+1

**Review by sachinpunadikar — APPROVED** [review]
+1

**Pull request merged**